### PR TITLE
kubemark: fix and enhance kubemark scripts for IKS

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind bug
/sig scalability
/sig ibmcloud
/cc @rtheis @kitch
/assign @shyamjvs @wojtek-t 

**What this PR does / why we need it**:

- fix shell script issues
  - `bx` is deprecated; rename to `ibmcloud`
  - remove unnecessary variable replacement in hollow-node_template.yaml
  - add replacement logic for HOLLOW_KUBELET_TEST_ARGS and HOLLOW_PROXY_TEST_ARGS
  - don't hardcode KUBEMARK_IMAGE_REGISTRY to brandondr96
- make cluster number and spec configurable
  - make number and spec of workers configurable
  - separate NUM_NODES and KUBEMARK_NUM_NODES

**Special notes for your reviewer**:

After #76848 gets merged, users can build their own kubemark image. At this moment, due to kubemark issue #69735, this PR can be verified:

```console
CLOUD_PROVIDER=iks [NUM_NODES=2] [KUBEMARK_NUM_NODES=10] ENABLE_KUBEMARK_CLUSTER_AUTOSCALER=N KUBEMARK_IMAGE_REGISTRY=hweicdl PROJECT=kubemark KUBEMARK_IMAGE_TAG=v1.12.7 test/kubemark/start-kubemark.sh
```

**Does this PR introduce a user-facing change?**:

```release-note
Kubemark scripts have been fixed for IKS clusters.
```

### CODE REVIEW

**Comment by wojtek-t** [comment]
@Huang-Wei - I'm happy to approve it once someone familiar with iks and ibmcloud will lgtm it.

**Review by rtheis — APPROVED** [review]
lgtm

File: cluster/kubemark/iks/config-default.sh [hunk 1]
```diff
@@ -25,8 +25,13 @@ CLUSTER_LOCATION="${CLUSTER_LOCATION:-wdc06}"
 REGISTRY_LOGIN_URL="${REGISTRY_LOGIN_URL:-https://api.ng.bluemix.net}"
 
 # User defined
+# number of real workers in spawnTester cluster
```
Comment (wojtek-t) [inline cluster/kubemark/iks/config-default.sh]: Nothing in kubemark should depend on NUM_NODES (as it shouldn't matter on how many nodes it has been run).

Instead, you should do the same that gce is doing:
NUM_NODES=${KUBEMARK_NUM_NODES:-10}
https://github.com/kubernetes/kubernetes/blob/master/cluster/kubemark/gce/config-default.sh#L29

Then you won't need a bunch of changes below too.

File: test/kubemark/iks/util.sh [hunk 2]
```diff
@@ -159,14 +164,14 @@ function complete-login {
...
+		ibmcloud login --sso -a "${REGISTRY_LOGIN_URL}"
 	elif [[ "${ISFED}" = "N" ]]; then
-		bx login -a "${REGISTRY_LOGIN_URL}"
+		ibmcloud login -a "${REGISTRY_LOGIN_URL}"
 	else
 		echo -e "${color_red}Invalid response, please try again:${color_norm}"
 		complete-login
 	fi
-	bx cr login
+	ibmcloud cr login
```
Comment (wojtek-t) [inline test/kubemark/iks/util.sh]: For the future - it would be useful to separate changes like s/bx/ibmcloud/ to a separate PR :-)

File: cluster/kubemark/iks/config-default.sh (hunk 1 above)
Comment (Huang-Wei) [reply cluster/kubemark/iks/config-default.sh]: Thanks for the prompt reivew! I read into the code, and noticed GCE works basically as below:

Firstly `--gcp-nodes` is passed to `kubetest`, and then it's passed onto env var:

```go
		{
			Env:    "NUM_NODES",
			Option: &o.gcpNodes,
			Name:   "--gcp-nodes",
		},
```

And then it uses bash (default) deployer to spin up a real cluster, which in turns calls `hack/e2e-internal/e2e-up.sh`:

```go
func (b *bashDeployer) Up() error {
	script := "./hack/e2e-internal/e2e-up.sh"
    ...
```

By this step, env var `NUM_NODES` literally means the number of the real cluster.

After that (the cluster is ready), `KUBEMARK_NUM_NODES` can overwrite the value of `NUM_NODES` (as the old value is not necessary any more), and we starts a hollow-node cluster as a rc.

```
NUM_NODES=${KUBEMARK_NUM_NODES:-10}
```

So basically GCE works in 2 phases: spin up a cluster and start the hollow-node rc. And they're working in serial, so we can overwrite NUM_NODES in latter phase with KUBEMARK_NUM_NODES.

For IKS, the embarrassing thing is `kubetest` doesn't support IKS yet, so `start-kubemark.sh` for IKS combines the 2 phases together. So IMO it's good to separate the 2 variables for IKS; otherwise it'd confuse people which literally means the number of the real cluster, and which means the number of the hollow-node rc.

File: test/kubemark/iks/util.sh (hunk 2 above)
Comment (Huang-Wei) [reply test/kubemark/iks/util.sh]: Absolutely. Will do next time.

File: cluster/kubemark/iks/config-default.sh (hunk 1 above)
Comment (wojtek-t) [reply cluster/kubemark/iks/config-default.sh]: OK - I didn't know you don't have support in kubetest.
In such case, it's probably the easiest thing to do to unblock you.

Can you please add a TODO with a comment why you do that and to change that once kubetest supports IKS?

Comment (Huang-Wei) [reply cluster/kubemark/iks/config-default.sh]: That makes sense.

Updated. PTAL.
